File: CustomizationEngine/lib/showcase/showcase_slots.dart
/// The slot manifest the engine's own **showcase** screens consume.
///
/// This exists so a consumer that only wants the preview (e.g. the
/// AppManagement admin app) can pass `ShowcaseSlots.expected*` into
/// `CustomizationRuntime.initialize` WITHOUT importing any host app's
/// private manifest (MobileApp's `CombatDenSlots`). The ids match the
/// pipeline's wire keys, so the same loaded customization resolves
/// identically here and in the member app.
///
/// It is intentionally a *subset* — only the slots the four showcase
/// screens actually render — so the engine's startup slot-validation
/// warnings stay scoped to what the preview shows.
class ShowcaseSlots {
  ShowcaseSlots._();

  // ---- Colour slots ----
  static const String primary = 'primary';
  static const String background = 'background';
  static const String text = 'text';
  static const String accent = 'accent';

  static const List<String> expectedColors = [
    primary,
    background,
    text,
    accent,
  ];

  // ---- Image slots ----
  static const String logoPrimary = 'logo_primary';
  static const String celebrationImage = 'celebration_image';
  static const String streakIcon = 'streak_icon';
  static const String giftbox = 'giftbox';
  static const String singlePoint = 'single_point';
  static const String pointsStarsImage = 'points_stars_image';
  static const String trophyImage = 'trophy_image';
  static const String rankBelt = 'rank_belt';
  static const String iconQrcode = 'icon_qrcode';

  static const List<String> expectedImages = [
    logoPrimary,
    celebrationImage,
    streakIcon,
    giftbox,
    singlePoint,
    pointsStarsImage,
    trophyImage,
    rankBelt,
    iconQrcode,
  ];

  // ---- Font slots ----
  static const String fontDisplay = 'display';
  static const String fontBody = 'body';

  static const List<String> expectedFonts = [
    fontDisplay,
    fontBody,
  ];

  // ---- Text slots ----
  static const String classBookedHeadline = 'class_booked_headline';
  static const String reserveCta = 'reserve_cta';
  static const String bookNextClassCta = 'book_next_class_cta';
  static const String winsTitle = 'wins_title';
  static const String winsSubtitle = 'wins_subtitle';

  static const List<String> expectedText = [
    classBookedHeadline,
    reserveCta,
    bookNextClassCta,
    winsTitle,
    winsSubtitle,
  ];

  // ---- Icon slots ----
  static const String navHome = 'nav_home';
  static const String navRank = 'nav_rank';
  static const String navReward = 'nav_reward';
  static const String navVideos = 'nav_videos';

  static const List<String> expectedIcons = [
    navHome,
    navRank,
    navReward,
    navVideos,
  ];

  // ---- Lottie slots ----
  static const String bookingCelebration = 'booking_celebration';
  static const String streakCelebration = 'streak_celebration';

  static const List<String> expectedLotties = [
    bookingCelebration,
    streakCelebration,
  ];
}

File: ThemeService/ThemeReact/src/app/showcase/showcaseSlots.ts
// Ports ../../../../../CRM/lib/showcase/showcase_slots.dart.
//
// The slot manifest the showcase screens consume — the ids the pipeline keys
// its wire payload by, so the same loaded theme resolves identically here and
// in the member app. It is deliberately a SUBSET of the member app's own
// manifest: only what the nine preview screens render, so the engine's
// startup missing-slot warning stays scoped to what the phone actually shows.
//
// These are the lists ../App.tsx hands to <ThemeProvider>'s `expected*` props
// (the analogue of Dart passing `ShowcaseSlots.expected*` into
// `ThemeRuntime.initialize`). Nothing else in the app declares slots.

// ---- Colour slots ----
export const SLOT_PRIMARY = 'primary';
export const SLOT_BACKGROUND = 'background';
export const SLOT_TEXT = 'text';
export const SLOT_ACCENT = 'accent';

export const EXPECTED_COLORS: readonly string[] = Object.freeze([
  SLOT_PRIMARY,
  SLOT_BACKGROUND,
  SLOT_TEXT,
  SLOT_ACCENT,
]);

// ---- Image slots ----
export const SLOT_LOGO_PRIMARY = 'logo_primary';
export const SLOT_CELEBRATION_IMAGE = 'celebration_image';
export const SLOT_STREAK_ICON = 'streak_icon';
export const SLOT_GIFTBOX = 'giftbox';
export const SLOT_SINGLE_POINT = 'single_point';
export const SLOT_POINTS_STARS_IMAGE = 'points_stars_image';
export const SLOT_TROPHY_IMAGE = 'trophy_image';
export const SLOT_RANK_BELT = 'rank_belt';
// The belt of the rank ABOVE the member's current one — the pipeline derives it
// from `rank_belt` (`apps/combatden/app.yaml`, `depends_on: rank_belt`). It is
// generated for every theme and read by the real member app
// (`MobileApp/lib/core/app_slots.dart`), whose Profile screen is its only
// consumer; the preview did not render Profile, so the slot was produced and
// then displayed nowhere.
export const SLOT_NEXT_RANK_BELT_IMAGE = 'next_rank_belt_image';
export const SLOT_ICON_QRCODE = 'icon_qrcode';

export const EXPECTED_IMAGES: readonly string[] = Object.freeze([
  SLOT_LOGO_PRIMARY,
  SLOT_CELEBRATION_IMAGE,
  SLOT_STREAK_ICON,
  SLOT_GIFTBOX,
  SLOT_SINGLE_POINT,
  SLOT_POINTS_STARS_IMAGE,
  SLOT_TROPHY_IMAGE,
  SLOT_RANK_BELT,
  SLOT_NEXT_RANK_BELT_IMAGE,
  SLOT_ICON_QRCODE,
]);

// ---- Font slots ----
export const SLOT_FONT_DISPLAY = 'display';
export const SLOT_FONT_BODY = 'body';

export const EXPECTED_FONTS: readonly string[] = Object.freeze([
  SLOT_FONT_DISPLAY,
  SLOT_FONT_BODY,
]);

// ---- Text slots ----
export const SLOT_CLASS_BOOKED_HEADLINE = 'class_booked_headline';
export const SLOT_RESERVE_CTA = 'reserve_cta';
export const SLOT_BOOK_NEXT_CLASS_CTA = 'book_next_class_cta';
export const SLOT_WINS_TITLE = 'wins_title';
export const SLOT_WINS_SUBTITLE = 'wins_subtitle';

export const EXPECTED_TEXT: readonly string[] = Object.freeze([
  SLOT_CLASS_BOOKED_HEADLINE,
  SLOT_RESERVE_CTA,
  SLOT_BOOK_NEXT_CLASS_CTA,
  SLOT_WINS_TITLE,
  SLOT_WINS_SUBTITLE,
]);

// ---- Icon slots ----
export const SLOT_NAV_HOME = 'nav_home';
export const SLOT_NAV_RANK = 'nav_rank';
export const SLOT_NAV_REWARD = 'nav_reward';
export const SLOT_NAV_VIDEOS = 'nav_videos';

export const EXPECTED_ICONS: readonly string[] = Object.freeze([
  SLOT_NAV_HOME,
  SLOT_NAV_RANK,
  SLOT_NAV_REWARD,
  SLOT_NAV_VIDEOS,
]);
